### PR TITLE
[PLGN-717] - Rapid7 InsightIDR-Allowing for a log id to be used instead of a log name in AdvancedQueryOnLog action

### PR DESCRIPTION
**Pull request by rbowden-r7**
### Description

Describe the proposed changes:

  - to allow for the use of either a log name or log id to be passed in from the ned user 
  - to bump the version of Aiohhtp used (https://github.com/rapid7/insightconnect-plugins/pull/2279)

https://rapid7.atlassian.net/browse/PLGN-717

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

new unit tests have been added to test the addition of the new paramater 

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

testing using local image and docker 

examples of logs from idr
image

fetched logs when using log name
image

fetched logs when using the log id
image

test for an invalid log
image

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

**Review by igorski-r7 — APPROVED** [review]

File: plugins/rapid7_insightidr/help.md [hunk 1]
```diff
@@ -3021,6 +3024,7 @@ Example output:
 
 # Version History
 
+* 9.0.0 - Actions: `Advanced Query On Log` - Now allows for ethier log id or log name to be used
```
Comment (dlaverty-r7) [inline plugins/rapid7_insightidr/help.md:3027]: Typo in either

File: plugins/rapid7_insightidr/komand_rapid7_insightidr/actions/advanced_query_on_log/action.py [hunk 2]
```diff
@@ -39,7 +40,17 @@ def run(self, params={}):
                 data=f"\nTime From: {time_from}\nTime To:{time_to}",
             )
 
-        log_id = self.get_log_id(log_name)
+        if log_id is None and log_name is None:
+            raise PluginException(
+                cause="No values were provided for either log id and log name",
```
Comment (dlaverty-r7) [inline plugins/rapid7_insightidr/komand_rapid7_insightidr/actions/advanced_query_on_log/action.py:45]: Suggestion reword: No values were provided for log id or log name

**Review by dlaverty-r7 — APPROVED** [review]
Lgtm just a couple of minor comments

File: plugins/rapid7_insightidr/komand_rapid7_insightidr/actions/advanced_query_on_log/action.py (hunk 2 above)
Comment (rbowden-r7) [reply plugins/rapid7_insightidr/komand_rapid7_insightidr/actions/advanced_query_on_log/action.py]: The logger message has been updated

**Pull request merged**